### PR TITLE
Nice, Corgi Pinning Action for the whole family.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Adds pin the tail on the corgi, a basic party game last seen at the summer ball, that I had lying around on one of my branches since then. The corgi is a poster that functions like most other posters, but has an interaction with the fake "Corgi tail" object, which can be pinned onto the poster to visually place the tail onto the poster.

Are you a bad enough dude to pin the tail on the corgi blindfolded?

## Why It's Good For The Game

Fun, cheap entertainment. Might make for a cheap thrill, or help set the scene of a birthday party. Works as a very player driven basic activity.

Also, the tail pins are harmless embeds. I'll leave that here for you to think about.
image

## Changelog
:cl:
add: The Games vendor now holds a box of "Pin the tail on the Corgi".
/:cl:

### CODE REVIEW

File: code/game/objects/items/tail_pin.dm [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	w_class = WEIGHT_CLASS_SMALL
+	throwforce = 0
+	throw_speed = 1
+	embedding = EMBED_HARMLESS
+	custom_materials = list(/datum/material/iron=1000)
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("poked", "jabbed", "pinned the tail on")
+	sharpness = IS_SHARP
+	max_integrity = 200
+	layer = 3.41 //Little trick to always stick itself above the poster.
```
Comment (Jared-Fogle) [inline code/game/objects/items/tail_pin.dm:15]: 😳

File: code/game/objects/items/tail_pin.dm [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/obj/structure/sign/poster/party_game
+	name = "pin the tail on the corgi"
+	desc = "The rules are simple, pin the tail on the corgi, while blindfolded. Are you a bad enough dude to hit the target?"
+	icon_state = "pinningposter"
+
+/obj/structure/sign/poster/party_game/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(istype(I,/obj/item/tail_pin))
+		if(user.a_intent != INTENT_HARM && !(I.item_flags & ABSTRACT)) //We're using the same trick that tables use for placing objects x and y onto the click location.
+			if(user.transferItemToLoc(I, drop_location(), silent = FALSE))
```
Comment (Jared-Fogle) [inline code/game/objects/items/tail_pin.dm:31]: Early returns

File: code/game/objects/items/tail_pin.dm [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		return
+	if(user.transferItemToLoc(I, drop_location(), silent = FALSE))
+		var/list/click_params = params2list(params)
+		if(!click_params || !click_params["icon-x"] || !click_params["icon-y"])
+			return
+		I.pixel_x = clamp(text2num(click_params["icon-x"]) - 16, -(world.icon_size/2), world.icon_size/2)
+		I.pixel_y = clamp(text2num(click_params["icon-y"]) - 16, -(world.icon_size/2), world.icon_size/2)
+		return TRUE
+	else
+		return ..()
```
Comment (Jared-Fogle) [inline code/game/objects/items/tail_pin.dm:42]: This should be an early return as well.

**Review by Mothblocks — DISMISSED** [review]

**Comment by ATH1909** [comment]
Not really sure what the point of making it an embed_harmless item is (meaning that it's only possible to embed it in a human by throwing it) when pinning the tail on the poster is only possible with a melee attack.

**Review by Twaticus — APPROVED** [review]

**Pull request merged**